tensorboard/webapp/metrics/store/metrics_types.ts

```diff
@@ -244,6 +247,7 @@ export interface MetricsNonNamespacedState {
   timeSeriesData: TimeSeriesData;
   isSettingsPaneOpen: boolean;
   isSlideoutMenuOpen: boolean;
+  tableEditorSelectedTab: DataTableMode;
```

Comment:
There is a strange fluidity between the word 'tab' and 'mode' throughout this PR. I tried to draw some lines but I am not sure they will make sense to others. Let me know if you have any suggestions.

Reply:
I did not find it problematic for the most part.

The one exception is one place where you seem to depend on the fact that DataTableModel.SINGLE is 0 and DataTableMode.RANGE is 1 and this happens to correspond with the tab indeces. I called this out in another comment.

Otherwise the relationship was clear to me.

Reply:
Glad to hear it. I explicitly made a function for returning the indeces.